heidrun/smithsonian.rb

```diff
@@ -0,0 +1,446 @@
+    uri record_uri
+  end
+
+  # dpla:SourceResource
+  sourceResource :class => DPLA::MAP::SourceResource do
+    # dcterms:isPartOf
+    #   <freetext category="setName" label="[n]">
+    collection :class => DPLA::MAP::Collection,
+               :each => record.field('freetext', 'setName'),
+               :as => :collection do
+      title collection
+    end
+
+    # dcterms:contributor
+    #  <name label="associated person">
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.field('freetext', 'name')
+                               .match_attribute(:label, 'associated person'),
+                :as => :contributor do
+      providedLabel contributor
+    end
+
+    # dcterms:creator
+    #   <freetext category="name" label="[value]">
+    creator :class => DPLA::MAP::Agent,
+            :each => record.field('freetext', 'name')
+                           .match_attribute(:label) { |label| creator?(label) },
+            :as => :creator do
+      providedLabel creator
+    end
+
+    # dc:date
+    #   <freetext category="date" label="[value]">
+    #   *Take earliest date
+    # Gretchen says (re only the earliest date):
+    #   So the mapping need to map all dates to both `sourceResource.date`
+    #   AND `sourceResource.temporal` and then we'll run enrichment on date
+    #   in the next step.
+    date :class => DPLA::MAP::TimeSpan,
+         :each => record.field('freetext', 'date'),
+         :as => :date do
+      providedLabel date
+    end
+
+    temporal :class => DPLA::MAP::TimeSpan,
+         :each => record.field('freetext', 'date'),
+         :as => :temporal do
+      providedLabel temporal
+    end
+
+    # dcterms:description
+    #   <freetext category="notes" label="[n]">[value]
+    #   *Each instance of "notes" should be a separate value.
+    description record.field('freetext', 'notes')
+
+    # dcterms:extent
+    #   <freetext category="physicalDescription" label="Dimensions">
+    extent record.field('freetext', 'physicalDescription')
+                 .match_attribute(:label, 'Dimensions')
+
+    # dc:format
+    #   <freetext category="physicalDescription" label="Physical description">
+    #   <freetext category="physicalDescription" label="Medium">;
+    #   <object_type>
+    #
+    # JB - actually I'm seeing this in indexedStructured
+    # but the spec doesn't seem to care about the parent anyway
+    dcformat record.map(&extract_format).flatten
+
+    # dcterms:identifier
+    #   <freetext category="identifier" label="Accession #">
+    #   <freetext category="identifier" label="Catalog #">
+    #   <record_ID>
+    identifier record.map(&identifier_map).flatten
+
+    # dcterms:language
+    #   <language> (not iso-6393 format)
+    # TODO: This will need an enrichment to convert to iso-6393
+    language record.field('indexedStructured', 'language').map(&clean_language)
+
+    # dcterms:spatial
+    #   <geoLocation><L5 type=[City | Town]></geoLocation >;
```

Comment:
Do you mean where coordinates, state, and city are all in separate properties? then yes...if you are asking something else, can you please describe?